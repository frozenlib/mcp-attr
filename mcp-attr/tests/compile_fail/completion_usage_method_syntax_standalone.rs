use mcp_attr::{Result, server::RequestContext};
use mcp_attr_macros::prompt;

#[prompt]
async fn test_prompt(#[complete(.method_name)] msg: String) -> Result<String> {
    Ok(msg)
}

fn main() {}