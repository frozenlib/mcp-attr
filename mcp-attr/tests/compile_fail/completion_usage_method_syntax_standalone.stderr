error: Method syntax (.method_name) can only be used within impl blocks, not in global functions
 --> tests/compile_fail/completion_usage_method_syntax_standalone.rs:5:34
  |
5 | async fn test_prompt(#[complete(.method_name)] msg: String) -> Result<String> {
  |                                  ^^^^^^^^^^^

warning: unused import: `server::RequestContext`
 --> tests/compile_fail/completion_usage_method_syntax_standalone.rs:1:24
  |
1 | use mcp_attr::{Result, server::RequestContext};
  |                        ^^^^^^^^^^^^^^^^^^^^^^
  |
  = note: `#[warn(unused_imports)]` on by default
